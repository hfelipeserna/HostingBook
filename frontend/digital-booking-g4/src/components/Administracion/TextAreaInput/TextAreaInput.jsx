import { Label } from "../../Forms/formElements"
import styles from "./TextAreaInput.module.css"

export default function TextAreaInput({
    label,
    name,
    placeholder,
    onChangeItem,
    setOnChangeItem
}) {
    return (
        <div className={styles.containerTextArea}>
            <label>{label}</label>
            <textarea
                name={name}
                id={name}
                cols="30"
                rows="6"
                value={onChangeItem}
                onChange={(e) => setOnChangeItem(e.target.value)}
                placeholder={placeholder}
            ></textarea>
        </div>
    )
}