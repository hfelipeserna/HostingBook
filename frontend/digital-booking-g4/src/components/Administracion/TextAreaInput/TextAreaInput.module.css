.containerTextArea{
    display: flex;
    flex-direction: column;
    width: 100%;
    
}

.containerTextArea label{
    color: var(--color-2);
    font-weight: 700;
    margin-top: 20px;
    cursor: pointer;
}

textarea{
    background-color: var(--color-5);
    border: 3px solid var(--color-5);
    border-radius: 5px;
    outline: none;
    box-shadow: 0px 1px 5px rgb(0 0 0 / 15%);
    color: var(--color-2);
    font-size: 1rem;
    padding: 1vh 40px 1vh 15px;
    width: 100%;
    margin-top: 5px;
    resize:none;
}

::placeholder {
    color: #BEBEBE;
    font-size: 0.75rem;
    font-weight: 500;
}