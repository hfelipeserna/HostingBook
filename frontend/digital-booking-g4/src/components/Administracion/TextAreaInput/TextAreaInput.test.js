import TextAreaInput from "./TextAreaInput"
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";


configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Text Area Input test", function () {
    const label = "este es el label"
    const name = "este es el name"
    const placeholder = "este es el placeHolder"
    let onChangeItem = ""
    const setOnChangeItem = (word) => {
        onChangeItem = word
    }

    test("Renderiza Por default", () => {
        const renderizacionSimple = render(<TextAreaInput></TextAreaInput>)
        expect(renderizacionSimple).toMatchSnapshot()

    });
    test("Renderiza Por default", () => {
        const renderizacion = mount(<TextAreaInput label={label} placeholder={placeholder} name={name} onChangeItem={onChangeItem} setOnChangeItem={setOnChangeItem}></TextAreaInput>)
        expect(renderizacion).toMatchSnapshot()
        const input = renderizacion.find('textarea')

        input.simulate('focus');
        input.simulate('change', { target: { value: 'a' } })
        input.simulate('keyDown', {
            which: 27,
            target: {
                blur() {
                    input.simulate('blur');
                },
            },
        });
        expect(onChangeItem).toEqual("a")

    });

});
