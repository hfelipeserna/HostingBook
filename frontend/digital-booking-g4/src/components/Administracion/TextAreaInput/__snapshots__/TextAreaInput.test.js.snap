// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Area Input test Renderiza Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerTextArea"
      >
        <label />
        <textarea
          cols="30"
          rows="6"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerTextArea"
    >
      <label />
      <textarea
        cols="30"
        rows="6"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text Area Input test Renderiza Por default 2`] = `
<div
  className="containerTextArea"
>
  <label>
    este es el label
  </label>
  <textarea
    cols="30"
    id="este es el name"
    name="este es el name"
    onChange={[Function]}
    placeholder="este es el placeHolder"
    rows="6"
    value=""
  />
</div>
`;
